cmd/cluster-agent/api/v1/languagedetection/util_test.go

```diff
@@ -454,13 +465,199 @@ func TestOwnersLanguagesMergeAndFlush(t *testing.T) {
+					},
+				},
+				dirty: false,
+			},
+			mockSupportedOwnerB: {
+				languages: langUtil.TimedContainersLanguages{
+					*langUtil.NewContainer("some-container"): {
+						"python": time.Now().Add(-5 * time.Minute),
+						"java":   time.Now().Add(-5 * time.Minute),
+					},
+				},
+				dirty: false,
+			},
+		},
+	}
+
+	ownersLanguages.cleanExpiredLanguages(mockStore)
+
+	assert.Eventuallyf(t,
+		func() bool {
+			deploymentA, err := mockStore.GetKubernetesDeployment("ns/deploymentA")
+			if err != nil {
+				return false
+			}
+
+			fmt.Println(deploymentA)
+
+			return reflect.DeepEqual(deploymentA.DetectedLanguages, langUtil.ContainersLanguages{
+				*langUtil.NewContainer("some-container"): {
+					"java": {},
+				},
+			})
+
+		},
+		2*time.Second,
+		100*time.Millisecond,
+		"Should find deploymentA in workloadmeta store with the correct languages")
+
+	assert.Eventuallyf(t,
+		func() bool {
+			_, err := mockStore.GetKubernetesDeployment("ns/deploymentB")
+			return err != nil
+		},
+		2*time.Second,
+		100*time.Millisecond,
+		"Should remove deploymentB from workloadmeta store since all languages are expired")
+
+}
+
+func TestCleanRemovedOwners(t *testing.T) {
+	mockSupportedOwnerA := langUtil.NewNamespacedOwnerReference("api-version", langUtil.KindDeployment, "deploymentA", "ns")
+
+	mockStore := fxutil.Test[workloadmeta.Mock](t, fx.Options(
+		core.MockBundle(),
+		fx.Supply(workloadmeta.NewParams()),
+		workloadmeta.MockModuleV2(),
+	))
+
+	ownersLanguages := OwnersLanguages{
+		containersLanguages: map[langUtil.NamespacedOwnerReference]*containersLanguageWithDirtyFlag{
+			mockSupportedOwnerA: {
+				languages: langUtil.TimedContainersLanguages{
+					*langUtil.NewContainer("some-container"): {
+						"python": time.Now().Add(5 * time.Minute),
+						"java":   time.Now().Add(5 * time.Minute),
+					},
+				},
+				dirty: false,
+			},
+		},
+	}
+
+	// start cleanup process
+	go ownersLanguages.cleanRemovedOwners(mockStore)
+
+	// Simulate detecting languages
+	mockStore.Push(workloadmeta.SourceLanguageDetectionServer, workloadmeta.Event{
+		Type: workloadmeta.EventTypeSet,
+		Entity: &workloadmeta.KubernetesDeployment{
+			EntityID: workloadmeta.EntityID{
+				Kind: workloadmeta.KindKubernetesDeployment,
+				ID:   "ns/deploymentA",
+			},
+			DetectedLanguages: langUtil.ContainersLanguages{
+				*langUtil.NewContainer("some-container"): {
+					"python": {},
+					"java":   {},
+				},
+			},
+		},
+	},
+	)
+
+	// simulate updating annotations
+	mockStore.Push("kubeapiserver", workloadmeta.Event{
+		Type: workloadmeta.EventTypeSet,
+		Entity: &workloadmeta.KubernetesDeployment{
+			EntityID: workloadmeta.EntityID{
+				Kind: workloadmeta.KindKubernetesDeployment,
+				ID:   "ns/deploymentA",
+			},
+			InjectableLanguages: langUtil.ContainersLanguages{
+				*langUtil.NewContainer("some-container"): {
+					"python": {},
+					"java":   {},
+				},
+			},
+		},
+	},
+	)
+
+	time.Sleep(1 * time.Second)
```

Comment:
it's better to use Eventually to avoid waiting more than necessary.

Reply:
Indeed

But I have just noticed that we don't need to wait here since we have another `Push` operation right after it and it is followed by `assert.Eventually`. 

Since order of handling events in wlm == order of pushing events, then no need to wait for the first event to be pushed.